spec.html

```diff
@@ -26297,8 +26356,173 @@ <h1>
+                [[Attributes]]
+              </td>
+              <td>
+                a List of ImportAttribute Records
+              </td>
+              <td>
+                The import attributes
+              </td>
+            </tr>
+            <tr>
+              <td>
+                [[Module]]
+              </td>
+              <td>
+                a Module Record
+              </td>
+              <td>
+                The loaded module corresponding to this module request
+              </td>
+            </tr>
+          </table>
+        </emu-table>
+
+        <p>An <dfn id="importattribute-record" variants="ImportAttribute Records">ImportAttribute Record</dfn> consists of the following fields:</p>
+        <emu-table id="table-importattribute-fields" caption="ImportAttribute Record Fields">
+          <table>
+            <tr>
+              <th>
+                Field Name
+              </th>
+              <th>
+                Value Type
+              </th>
+              <th>
+                Meaning
+              </th>
+            </tr>
+            <tr>
+              <td>
+                [[Key]]
+              </td>
+              <td>
+                a String
+              </td>
+              <td>
+                The attribute key
+              </td>
+            </tr>
+            <tr>
+              <td>
+                [[Value]]
+              </td>
+              <td>
+                a String
+              </td>
+              <td>
+                The attribute value
+              </td>
+            </tr>
+          </table>
+        </emu-table>
+
+        <emu-clause id="sec-ModuleRequestsEqual" type="abstract operation">
+          <h1>
+            ModuleRequestsEqual (
+              _left_: a ModuleRequest Record or a LoadedModuleRequest Record,
+              _right_: a ModuleRequest Record or a LoadedModuleRequest Record,
+            ): a Boolean
+          </h1>
+          <dl class="header">
+            <dt>description</dt>
+            <dd></dd>
+          </dl>
+
+          <emu-alg>
+            1. If _left_.[[Specifier]] is not _right_.[[Specifier]], return *false*.
+            1. Let _leftAttrs_ be _left_.[[Attributes]].
+            1. Let _rightAttrs_ be _right_.[[Attributes]].
+            1. Let _leftAttrsCount_ be the number of elements in _leftAttrs_.
+            1. Let _rightAttrsCount_ be the number of elements in _rightAttrs_.
+            1. If _leftAttrsCount_ ≠ _rightAttrsCount_, return *false*.
+            1. For each ImportAttribute Record _l_ of _leftAttrs_, do
+              1. Let _found_ be *false*.
+              1. For each ImportAttribute Record _r_ of _rightAttrs_, do
+                1. If _l_.[[Key]] is _r_.[[Key]], then
+                  1. If _l_.[[Value]] is _r_.[[Value]], then
+                    1. Assert: _found_ is *false*.
+                    1. Set _found_ to *true*.
+                  1. Else,
+                    1. Return *false*.
```

Comment:
Given that we have searches elsewhere in the spec of the form "If _list_ does not contain a FooRecord such that <some property holds>, ...", I'd prefer that form to be used here for readability instead of a manual algorithmic search.

Reply:
Changed to
```
1. For each ImportAttribute Record _l_ of _leftAttrs_, do
  1. If _rightAttrs_ does not contain an ImportAttribute Record _r_ such that _l_.[[Key]] is _r_.[[Key]] and _l_.[[Value]] is _r_.[[Value]], return *false*.
1. Return *true*.
 ```

Reply:
That reads great, thanks!